tool/tctl/common/plugin/plugins_command.go

```diff
@@ -311,22 +312,42 @@ func (p *PluginsCommand) InstallSCIM(ctx context.Context, client *authclient.Cli
 }
 
 // TryRun runs the plugins command
-func (p *PluginsCommand) TryRun(ctx context.Context, cmd string, client *authclient.Client) (match bool, err error) {
+func (p *PluginsCommand) TryRun(ctx context.Context, cmd string, clientFunc commonclient.InitFunc) (match bool, err error) {
+	var commandFunc func(ctx context.Context, client *authclient.Client) error
 	switch cmd {
 	case p.cleanupCmd.FullCommand():
-		err = p.Cleanup(ctx, client)
+		commandFunc = p.Cleanup
 	case p.install.okta.cmd.FullCommand():
+		client, closeFn, err := clientFunc(ctx)
+		if err != nil {
+			return false, trace.Wrap(err)
+		}
+		defer closeFn(ctx)
 		args := installPluginArgs{authClient: client, plugins: client.PluginsClient()}
 		err = p.InstallOkta(ctx, args)
+		return true, trace.Wrap(err)
 	case p.install.scim.cmd.FullCommand():
-		err = p.InstallSCIM(ctx, client)
+		commandFunc = p.InstallSCIM
 	case p.install.entraID.cmd.FullCommand():
+		client, closeFn, err := clientFunc(ctx)
+		if err != nil {
+			return false, trace.Wrap(err)
+		}
+		defer closeFn(ctx)
 		args := installPluginArgs{authClient: client, plugins: client.PluginsClient()}
 		err = p.InstallEntra(ctx, args)
+		return true, trace.Wrap(err)
 	case p.delete.cmd.FullCommand():
-		err = p.Delete(ctx, client)
+		commandFunc = p.Delete
 	default:
 		return false, nil
 	}
```

Comment:
Using a mix of early return and setting command func is a bit confusing.
Can you change the Delete, Cleanup and SCIM commands to use the same `installPluginArgs` so you can have a single commandFunc invocation for everything
```
	var commandFunc func(ctx context.Context, pluginArgs installPluginArgs) error
```

I think you might have to add a few methods to their `authClient` interface (at least `GetRole` for InstallSCIM).